src/core/connection.c

```diff
@@ -3644,14 +3642,12 @@ QuicConnRecvPrepareDecrypt(
             CXPLAT_DBG_ASSERT(Connection->Crypto.TlsState.ReadKeys[QUIC_PACKET_KEY_1_RTT_OLD] != NULL);
             CXPLAT_DBG_ASSERT(Connection->Crypto.TlsState.WriteKeys[QUIC_PACKET_KEY_1_RTT_OLD] != NULL);
             Packet->KeyType = QUIC_PACKET_KEY_1_RTT_OLD;
-        } else {
+        } else if (Packet->PacketNumber > PacketSpace->ReadKeyPhaseEndPacketNumber) {
```

Comment:
I think we should do `>=` in case we receive the a duplicate packet. I don't want to mark it as a decryption failure eventually.

Reply:
>= is incorrect per the spec. If this is hit, we've already received that packet in the current key space. There is NO scenario that packet can be legal ever. The only cause of it would be a bad implementation or a malicious peer.

Reply:
The main reason is its not a fully duplicate packet. Its a duplicate packet with the incorrect key phase if its ever in range inclusive. In that case, we have 0 clue which key would even potentially decode it.

Reply:
Ok. Makes sense.